asserts/asserts.go

```diff
@@ -29,22 +29,40 @@ import (
 	"unicode/utf8"
 )
 
-// AssertionType labels assertions of a given type
-type AssertionType string
+// AssertionType describes a known assertion type with its name and metadata.
+type AssertionType struct {
+	// Name of the type.
+	Name string
+	// PrimaryKey holds the names of the headers that constitute the
+	// unique primary key for this assertion type.
+	PrimaryKey []string
+	assembler  func(assert assertionBase) (Assertion, error)
```

Comment:
At least one line separating this odd private case from the others would be nice. I'd probably add more myself between the fields above, but that's personal preference.